editor/editor_inspector.cpp

```diff
@@ -3114,7 +3114,24 @@ void EditorInspector::update_tree() {
 			// Build the doc hint, to use as tooltip.
 
 			// Get the class name.
-			StringName classname = doc_name == "" ? object->get_class_name() : doc_name;
+			StringName classname = doc_name;
+			if (classname == "") {
+				if (Object::cast_to<Resource>(object) && object->get_script()) {
+					Resource *res = cast_to<Resource>(object);
+					if (res) {
```

Comment:
Can this ever be false? To get inside `if (Object::cast_to<Resource>(object) && object->get_script()) {`, there had to have been a successful cast into `Resource`, so `res` can never be `nullptr`.

Looks like the ifs should be reorganized a little bit?

```
if (classname == "") {
  Resource *res = cast_to<Resource>(object);
  if (res) {
    Script *src = cast_to<Script>(res->get_script());
    if (src) { ... } else { ... }
```

Something like this?

Reply:
After defaulting class_name to the object's class name, as you suggested, there is no more need for else conditions. Thanks.